unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+
+    controller_set_surface_source_rectangle(nullptr, nullptr, 1, 1, 1, 1, 1);
+
+    ASSERT_EQ(surface_set_source_rectangle_fake.call_count, 1);
+    ASSERT_EQ(surface_set_source_rectangle_fake.arg1_val, 1);
+    ASSERT_EQ(surface_set_source_rectangle_fake.arg2_val, 1);
+    ASSERT_EQ(surface_set_source_rectangle_fake.arg3_val, 1);
+    ASSERT_EQ(surface_set_source_rectangle_fake.arg4_val, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_set_surface_source_rectangle_successWithNegativeValues
+ * @brief               Test case of controller_set_surface_source_rectangle() where get_surface_from_id() success, return an object
+ *                      and input x, y, width, height are negative values
+ * @test_procedure Steps:
+ *                      -# Set data of ivi_layout_surface_properties to default value {0}
+ *                      -# Mocking the get_properties_of_surface() does return an object
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Mocking the get_surface_from_id() does return an object
+ *                      -# Calling the controller_set_surface_source_rectangle()
+ *                      -# Verification point:
+ *                         +# surface_set_source_rectangle() must be called once time
+ *                         +# Input data of surface_set_source_rectangle() must same default value of ivi_layout_surface_properties
+ */
+TEST_F(ControllerTests, controller_set_surface_source_rectangle_successWithNegativeValues)
+{
+    const struct ivi_layout_surface_properties l_prop = {
+        .opacity = 0,
+        .source_x = 0,
+        .source_y = 0,
+        .source_width = 0,
+        .source_height = 0,
+        .start_x = 0,
+        .start_y = 0,
+        .start_width = 0,
+        .start_height = 0,
+        .dest_x = 0,
+        .dest_y = 0,
+        .dest_width = 0,
+        .dest_height = 0,
+        .orientation = WL_OUTPUT_TRANSFORM_NORMAL,
+        .visibility = true,
+        .transition_type = 0,
+        .transition_duration = 0,
+        .event_mask = 0,
+    };
+    const struct ivi_layout_surface_properties *prop[1] = {&l_prop};
+
+    SET_RETURN_SEQ(get_properties_of_surface, prop, 1);
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviController, 1);
+    struct ivi_layout_surface *l_layout_surface[1] = {(struct ivi_layout_surface *)0xFFFFFFFF};
+    SET_RETURN_SEQ(get_surface_from_id, l_layout_surface, 1);
+
+    controller_set_surface_source_rectangle(nullptr, nullptr, 1, -1, -1, -1, -1);
+
+    ASSERT_EQ(surface_set_source_rectangle_fake.call_count, 1);
+    ASSERT_EQ(surface_set_source_rectangle_fake.arg1_val, 0);
+    ASSERT_EQ(surface_set_source_rectangle_fake.arg2_val, 0);
+    ASSERT_EQ(surface_set_source_rectangle_fake.arg3_val, 0);
+    ASSERT_EQ(surface_set_source_rectangle_fake.arg4_val, 0);
```

Comment:
should verify like:
    ASSERT_EQ(surface_set_source_rectangle_fake.call_count, 1);
    ASSERT_EQ(surface_set_source_rectangle_fake.arg1_val, l_prop.source_x);
    ASSERT_EQ(surface_set_source_rectangle_fake.arg2_val, l_prop.source_y);
    ASSERT_EQ(surface_set_source_rectangle_fake.arg3_val, l_prop.source_width);
    ASSERT_EQ(surface_set_source_rectangle_fake.arg4_val, l_prop.source_height);